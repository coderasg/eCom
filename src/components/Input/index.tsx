import React, { ChangeEvent } from 'react';

type Props = {
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  placeholder: string;
  className?: string;
  name: string;
  value: string;
};

function Input(props: Props) {
  const { onChange, name,value,  className, placeholder } = props;

  return (
    <div data-testid='Input-wrapper' className={className}>
      <input
        data-testid={'searchInput'}
        value={value}
        type='text'
        name={name}
        placeholder={placeholder}
        onChange={onChange}
      />
    </div>
  );
}

export default Input;
