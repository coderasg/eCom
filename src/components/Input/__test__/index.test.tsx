import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../index';

describe('(Component) Input', () => {
  const defaultProps = {
    value: 'searc..',
    name:'searchProduct',
    placeholder:'Search product',
    onChange:()=> {
      console.log('Input chnage');
    },
  };
  
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<Input {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
