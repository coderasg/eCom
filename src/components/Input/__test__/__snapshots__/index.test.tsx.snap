// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Input should pass snapshot test 1`] = `
<div
  data-testid="Input-wrapper"
>
  <input
    data-testid="searchInput"
    name="searchProduct"
    onChange={[Function]}
    placeholder="Search product"
    type="text"
    value="searc.."
  />
</div>
`;
